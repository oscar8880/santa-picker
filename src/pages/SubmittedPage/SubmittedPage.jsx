import styles from './SubmittedPage.module.scss';
import React, { useState, useEffect, useContext } from 'react';
import Text from '../../components/Text/Text';
import Action from '../../components/Action/Action';
import PageWrapper from '../../components/PageWrapper/PageWrapper';
import { Container, Row, Col } from 'react-grid-system';
import sleighImage from '../../images/sleigh.png';
import { useSpring, animated, config } from 'react-spring';
import { ParticipantContext } from '../../context/ParticipantContext';
import { simpleCypher } from '../../utils/simpleCypher';

const SubmittedPage = () => {
  const [loaded, setLoaded] = useState(false);
  useEffect(() => setLoaded(true), []);
  const { participantsData } = useContext(ParticipantContext);

  const resultLinks = participantsData.contacts.map((contacts) => {
    return `${participantsData.spendingLimit || '_'}Y${simpleCypher(
      contacts.contact.name,
    )}J${simpleCypher(contacts.assignedContact.name)}T${simpleCypher(
      participantsData.organiserName,
    )}`;
  });

  const contentProps = useSpring({
    config: config.gentle,
    opacity: loaded ? 1 : 0,
    marginTop: loaded ? 0 : -70,
  });

  return (
    <div className={styles.SubmittedPage}>
      <PageWrapper>
        {loaded && (
          <animated.div style={contentProps}>
            <Container fluid>
              <Row>
                <Col xs={12} md={6} lg={6}>
                  <div className={styles.SubmittedPage__StrapLine}>
                    <Text appearance="h6" tagName="h1">
                      Success!
                    </Text>
                  </div>
                  <ul className={styles.SubmittedPage__BlurbTextList}>
                    <li className={styles.SubmittedPage__BlurbTextListItem}>
                      <Text appearance="h5">
                        Emails are on their way to your Secret Santas
                      </Text>
                    </li>
                    <li className={styles.SubmittedPage__BlurbTextListItem}>
                      <Text appearance="h5">
                        The courier elves might get lost so check your spam
                        folder if they&apos;re taking a while
                      </Text>
                    </li>
                    <li className={styles.SubmittedPage__BlurbTextListItem}>
                      <Text appearance="h5">
                        Alternatively send your santas their personalised link
                        below
                      </Text>
                    </li>
                  </ul>
                  {participantsData.contacts
                    .sort((a, b) => {
                      return a.contact.name < b.contact.name ? -1 : 1;
                    })
                    .map((contact, i) => {
                      return (
                        <>
                          <p key={contact.contact.name}>
                            {contact.contact.name} has been assigned{' '}
                            {contact.assignedContact.name}
                          </p>
                          <Action
                            tagName="Link"
                            to={encodeURI(`/result/${resultLinks[i]}`)}
                          >
                            Link
                          </Action>
                        </>
                      );
                    })}
                  <div className={styles.SubmittedPage__CTA}>
                    <Action tagName="Link" to="/" appearance="button-secondary">
                      Back
                    </Action>
                  </div>
                </Col>
                <Col xs={12} md={6} lg={5} offset={{ lg: 1 }}>
                  <div className={styles.SubmittedPage__Image}>
                    <img
                      src={sleighImage}
                      alt="Sleigh"
                      width="80%"
                      height="80%"
                    ></img>
                  </div>
                </Col>
              </Row>
            </Container>
          </animated.div>
        )}
      </PageWrapper>
    </div>
  );
};

export default SubmittedPage;
