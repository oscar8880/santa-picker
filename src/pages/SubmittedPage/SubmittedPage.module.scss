@import "../../scss/functions";
@import "../../scss/variables";

.SubmittedPage {
  display: flex;
  flex-direction: column;
  padding-bottom: $spacer-12;
  padding-top: $spacer-10;
  @media (min-width: $breakpoint-md) {
    padding-top: $spacer-12;
  }
}

.SubmittedPage__StrapLine {
  padding-bottom: $spacer-8;
  text-align: center;
  @media (min-width: $breakpoint-md) {
    text-align: left;
  }
}

.SubmittedPage__BlurbTextList {
  list-style: none;
  padding-bottom: $spacer-4;
  text-align: center;
  @media (min-width: $breakpoint-md) {
    text-align: left;
  }
}

.SubmittedPage__BlurbTextListItem {
  color: $grey-500;
  padding-bottom: $spacer-6;
}

.SubmittedPage__Name {
  color: $grey-800;
}

.SubmittedPage__CopyButton {
  padding-left: 0;
  padding-right: 0;
  width: 60px;
}

.SubmittedPage__LinkContainer {
  display: flex;
  margin-top: $spacer-2;
  width: 100%;
}

.SubmittedPage__TextArea {
  background-color: $grey-100;
  border: none;
  color: $grey-300;
  font-size: $font-size-h5;
  height: 47px;
  padding-bottom: 0;
  padding-left: $spacer-4;
  padding-right: $spacer-4;
  padding-top: 11px;
  resize: none;
  white-space: nowrap;
  width: 100%;
}

.SubmittedPage__Image {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  padding-top: $spacer-8;
  width: 80%;

  @media (min-width: $breakpoint-md) {
    justify-content: flex-start;
    padding-top: $spacer-9;
    width: 100%;
  }
}

.SubmittedPage__CTA {
  padding-bottom: $spacer-12;
  padding-top: $spacer-4;
}
