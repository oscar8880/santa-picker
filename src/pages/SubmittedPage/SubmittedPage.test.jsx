import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';

import SubmittedPage from './SubmittedPage';

test('matches snapshot', async () => {
  const { asFragment } = render(
    <MemoryRouter>
      <SubmittedPage />
    </MemoryRouter>,
  );

  expect(asFragment()).toMatchSnapshot();
});
