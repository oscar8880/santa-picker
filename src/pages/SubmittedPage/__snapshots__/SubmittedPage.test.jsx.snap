// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SubmittedPage"
  >
    <div
      class="PageWrapper"
    >
      <div
        style="box-sizing: border-box; position: relative; margin-left: auto; margin-right: auto; padding-left: 15px; padding-right: 15px;"
      >
        <div
          style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
        >
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;"
          >
            <div
              class="SubmittedPage__StrapLine"
            >
              <h1
                class="Text Text__h6"
              >
                Success!
              </h1>
            </div>
            <ul
              class="SubmittedPage__BlurbTextList"
            >
              <li
                class="SubmittedPage__BlurbTextListItem"
              >
                <p
                  class="Text Text__h5"
                >
                   Emails are on their way to your Secret Santas
                </p>
              </li>
              <li
                class="SubmittedPage__BlurbTextListItem"
              >
                <p
                  class="Text Text__h5"
                >
                  The courrier elves might get lost so check your spam folder if they're taking a while
                </p>
              </li>
            </ul>
            <div
              class="SubmittedPage__CTA"
            >
              <a
                class="Action Action__Appearance--button-secondary"
                href="/"
              >
                Home
              </a>
            </div>
          </div>
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 41.66666666666667%; flex-basis: 41.66666666666667%; flex-grow: 0; flex-shrink: 0; max-width: 41.66666666666667%; margin-left: 8.333333333333334%;"
          >
            <div
              class="SubmittedPage__Image"
            >
              <img
                alt="Sleigh"
                src="sleigh.png"
                width="80%"
              />
            </div>
          </div>
        </div>
        <span
          style="display: table; clear: both;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
